src/nebula-hooks/use-apply-column-widths.ts

```diff
@@ -0,0 +1,29 @@
+import { useMemo } from '@nebula.js/stardust';
+import { HyperCube, ApplyColumnWidths } from '../types';
+
+export const applyColumnWidthsFactory =
+  (model: EngineAPI.IGenericObject | undefined, qHyperCube: HyperCube): ApplyColumnWidths =>
+  (newColumnWidth, { isDim, colIdx }) => {
+    const index = isDim ? colIdx : colIdx - qHyperCube.qDimensionInfo.length;
+    const qPath = `/qHyperCubeDef/${isDim ? 'qDimensions' : 'qMeasures'}/${index}/qDef/columnWidth`;
+    const oldColumnWidth = qHyperCube[isDim ? 'qDimensionInfo' : 'qMeasureInfo'][index].columnWidth;
+
+    const patch = oldColumnWidth
+      ? {
+          qPath,
+          qOp: 'Replace' as EngineAPI.NxPatchOpType,
+          qValue: JSON.stringify({ ...oldColumnWidth, ...newColumnWidth }),
+        }
+      : {
+          qPath,
+          qOp: 'Add' as EngineAPI.NxPatchOpType,
+          qValue: JSON.stringify(newColumnWidth),
+        };
+
+    model?.applyPatches([patch], true);
```

Comment:
it doesn't return undefined. the factory always returns `ApplyColumnWidths`. it is just that the if the model is undefined, we don't call `applyPatches`.  Which will never happen, since when you can actually change the width, model will be defined.
The reason why `applyColumnWidths` can be undefined in root, is because it is only passed to root in the pagination case. In the other cases it will be undefined

Reply:
We will sort out this typing mess when we get a decision about if we can remove the mobile stuff, then things will make more sense